a10_neutron_lbaas/v2/handler_listener.py

```diff
@@ -42,19 +48,39 @@ def _set(self, set_method, c, context, listener):
 
         templates = self.meta(listener, "template", {})
 
+        bindings = []
         server_args = {}
         cert_data = dict()
         template_args = {}
 
+        # Try Barbican first.  TERMINATED HTTPS requires a default TLS container ID that is
+        # checked by the API so we can't fake it out.
         if listener.protocol and listener.protocol == constants.PROTOCOL_TERMINATED_HTTPS:
             if self._set_terminated_https_values(listener, c, cert_data):
                 templates["client_ssl"] = {}
-                template_name = str(cert_data.get('template_name', ''))
-                key_passphrase = str(cert_data.get('key_pass', ''))
-                cert_filename = str(cert_data.get('cert_filename', ''))
-                key_filename = str(cert_data.get('key_filename', ''))
+                template_name = str(cert_data.get('template_name') or '')
+                key_passphrase = str(cert_data.get('key_pass') or '')
+                cert_filename = str(cert_data.get('cert_filename') or '')
+                key_filename = str(cert_data.get('key_filename') or '')
             else:
                 LOG.error("Could not created terminated HTTPS endpoint.")
+        # Else, set it up as an HTTP endpoint and attach the A10 cert data.
+        elif (self.a10_driver.config.get('use_database') and listener.protocol and
+              listener.protocol in [constants.PROTOCOL_HTTP,
+                                    constants.PROTOCOL_TCP]):
```

Comment:
Why is this only HTTP and TCP? HTTP doesn't work at all on my ax, I think it needs to be HTTPS, but that doesn't work with an HTTP pool ...